packages/manager/modules/billing/src/components/utils/billing.service.js

```diff
@@ -36,9 +36,21 @@ export default class BillingService {
       .then((commitments) => commitments.map((c) => new Commitment(c)));
   }
 
+  getEngagementDetails(serviceId) {
+    return this.$http
+      .get(`/services/${serviceId}/billing/engagement`)
+      .then(({ data }) => data);
+  }
+
```

Comment:
I think there are info about engagement in the `/services` API maybe we might not need another call